### ISSUE TITLE
Static Analysis

### ISSUE THREAD

**Issue by JayWoodroffe** [BME-MIT-IET/ivt-ha-2024-pied-piper-1#1]
Running a static analysis tool and reviewing the reported problems (SonarQube, FindBugs, VS Code Analyzer, Codacy, Coverity Scan...). As static analysis tools could find numerous problems, it is enough to review only a subset of them. If the team agrees with the reported problem, it can fix some of them. Try to review different types of problem and concentrate on more complex ones (e.g. not just formatting errors).

**Comment by punketeee** [comment]
I ran Lint as a static analysis tool and here is a list of some of the reported problems from the results. Some of the reported problems are obsolete layout params, overdrawing, long vector paths, hardcoded text, image without content description, query permissions needed. Can you check out the last 3?

**Comment by JayWoodroffe** [comment]
> I ran Lint as a static analysis tool. Some of the reported problems are obsolete layout params, overdrawing, long vector paths, hardcoded text, image without content description, query permissions needed. Can you check out the last 3?

Absolutely. I can look through them tonight and summarise the issues I find by tomorrow morning.

**Comment by punketeee** [comment]
After a meeting, we have decided for my colleague to leave out image without content description error and instead focus on implied default locale in case conversion error. 

List of errors we have reviewed: 

**Performance Issues:**
- Obsolete Layout Params issue 
- Overdraw issue
- Long Vector Paths issue

**Correctness Issues:**
- Using APIs affected by query permissions
- Implied default locale in case conversion

**Internationalization Issues:**
- Hardcoded text 

1. Obsolete Layout Params
![image](https://github.com/BME-MIT-IET/ivt-ha-2024-pied-piper-1/assets/144776564/7dbc29d3-862a-404d-a66c-2199e1bb4964)

In the file “activity_register_contact.xml”, there are many examples of using obsolete practices. While using LinearLayout, android:layout_below is utilized but this is incorrect: layout_below is only applicable in RelativeLayout. 
-	Solution suggestion: attributes compatible with LinearLayout can be used (e.g. android:layout_marginTop).


2. Overdraw
![image](https://github.com/BME-MIT-IET/ivt-ha-2024-pied-piper-1/assets/144776564/b1903dcc-edb8-44f9-b484-fc6aa08df564)

In the file activity_apoointments.xml, overdraw is a repeating problem. Overdraw happens when a screen region is painted multiple times and it occurs when both the root layout and the theme apply the same background causing performance issues and unnecessary GPU usage. 

3. Long Vector Paths 
![image](https://github.com/BME-MIT-IET/ivt-ha-2024-pied-piper-1/assets/144776564/40e4eb4c-b4cc-4422-9c04-4aa398cc59a8)

In ic_settings.xml, there is a 904 character long vector path. Vectors with such complex paths can impact app performance due to high processing needs. It can cause lag when drawing views or even app crashes if the vector drawing consumes too much memory or resources. 
Solution alternatives:
-	Vector path can be simplified by reducing precision or eliminating excessive details.
-	Complex vectors can be rasterized into bitmaps (PNG) for better performance.

**Comment by JayWoodroffe** [comment]
4. Using APIs affected by query permissions
![image](https://github.com/BME-MIT-IET/ivt-ha-2024-pied-piper-1/assets/77754204/f1bd9d34-a2ce-4db5-8b0f-b1d2b780fae6)

In ClientDetails.kt, an API that is affected by query permissions is used. Recent Android versions require apps to explicitly declare other apps or packages they intend to query using intents. Without a <queries> permission, intent resolutions might fail. 

5. Implied default locale in case conversion
![image](https://github.com/BME-MIT-IET/ivt-ha-2024-pied-piper-1/assets/77754204/fff719a0-a95c-4a13-b06f-09ed18efb5c4)

If toLowerCase() or toUpperCase() methods are used without specifying locale, the behavior can vary across devices due to different locales, thus leading to incorrect data handling, especially for languages other than English. This issue is present in AddClient.kt. 

6. Hardcoded text
![image](https://github.com/BME-MIT-IET/ivt-ha-2024-pied-piper-1/assets/77754204/5e6beabc-de75-4652-a333-554ce408f395)

In documents such as activity_add_appointment.xml, activity_add_client.xml, activity_appointments.xml, etc. there is a recurring issue: the text strings in the layout are directly specified, which results in them not being translated and makes it difficult to localize the app for different regions.

**Comment by punketeee** [comment]
commit 53ed53b76fcf1d198eefdd9ad086fc1987650a49
